lib/main.dart

```diff
@@ -62,26 +66,47 @@ class MainApp extends StatefulWidget {
           boxCommunicator.downloadData(context);
           break;
         } else if (name != null) {
-          boxConnection.connectedToWifi();
-          boxCommunicator.uploadToBackend(context);
+          // connected to a wifi
+          if (storage.get('knownWifis', defaultValue: []).contains(name)) {
+            boxConnection.connectedToKnownWifi();
+            boxCommunicator.uploadToBackend(context);
+          } else {
+            boxConnection.connectedToUnknownWifi();
+          }
         }
         break;
 
       case Connection.SENSORBOX:
         if (name == null)
           boxConnection.disconnected();
         else if (name != "Sensorbox") {
-          boxConnection.connectedToWifi();
-          boxCommunicator.uploadToBackend(context);
+          if (storage.get('knownWifis', defaultValue: []).contains(name)) {
```

Comment:
store storage.get('knownWifis', defaultValue: []) in a List so that you dont have to call it twice